warden-roboto/src/main/kotlin/AndroidAttestationChecker.kt

```diff
@@ -273,40 +273,57 @@ abstract class AndroidAttestationChecker(
     @Throws(AttestationValueException::class)
     protected abstract fun ParsedAttestationRecord.verifyAndroidVersion(
         versionOverride: Int? = null,
-        osPatchLevel: PatchLevel?
+        osPatchLevel: PatchLevel?,
+        verificationDate: Date
     )
 
-    protected fun AuthorizationList.verifyAndroidVersion(versionOverride: Int?, patchLevel: PatchLevel?) {
-        runCatching {
+    protected fun AuthorizationList.verifyAndroidVersion(
+        versionOverride: Int?,
+        patchLevel: PatchLevel?,
+        verificationDate: Date
+    ) = catchingUnwrapped {
+        (versionOverride ?: attestationConfiguration.androidVersion)?.let {
+            if ((osVersion().get()) < it) throw AttestationValueException(
+                "Android version not supported: ${osVersion().get()} (should be at least $it)",
+                reason = AttestationValueException.Reason.OS_VERSION,
+                expectedValue = it,
+                actualValue = osVersion().get()
+            )
+        }
 
-            (versionOverride ?: attestationConfiguration.androidVersion)?.let {
-                if ((osVersion().get()) < it) throw AttestationValueException(
-                    "Android version not supported: ${osVersion().get()} (should be at least $it)",
-                    reason = AttestationValueException.Reason.OS_VERSION,
-                    expectedValue = it,
-                    actualValue = osVersion().get()
-                )
-            }
+        (patchLevel ?: attestationConfiguration.patchLevel)?.let {
+            if ((osPatchLevel().get()).isBefore(YearMonth.of(it.year, it.month))) throw AttestationValueException(
+                "Patch level not supported: ${osPatchLevel().get()} (should be at least $it)",
+                reason = AttestationValueException.Reason.OS_VERSION,
+                expectedValue = it,
+                actualValue = osPatchLevel().get()
+            )
+        }
 
-            (patchLevel ?: attestationConfiguration.patchLevel)?.let {
-                if ((osPatchLevel().get()).isBefore(YearMonth.of(it.year, it.month))) throw AttestationValueException(
-                    "Patch level not supported: ${osPatchLevel().get()} (should be at least $it)",
+        (patchLevel ?: attestationConfiguration.patchLevel)?.let {
+            it.maxFuturePatchLevelMonths?.let { maxFuturePatchLevelMonths ->
+                val fromAttestation = osPatchLevel().get()
+                val calendar = Calendar.getInstance().apply { time = verificationDate }
+                val currentYearMonth = YearMonth.of(calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH))
```

Comment:
Java's Calendar has the months 0-based
```suggestion
                val currentYearMonth = YearMonth.of(calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH) + 1)
```

Reply:
i would never have thought about this. thanks!